node/api/middleware.go

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2020 The virtual-kubelet authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"net/http"
+)
+
+// Middleware defines a middleware that is to
+// take one HandlerFunc and wrap it within another HandlerFunc
+type Middleware func(http.HandlerFunc) http.HandlerFunc
```

Comment:
@sargun @cpuguy83 currently there is no handler constructed in node-cli, it calls [api.AttachPodRoutes(podRoutes, mux, true)](https://github.com/virtual-kubelet/node-cli/blob/master/internal/commands/root/http.go#L128) which is inside of virtual-kubelet/virtual-kubelet. As far as I know, a middleware can wrap Handler or HandlerFunc, but it can't wrap a ServerMux, so it looks like we need to put the middleware here. But we might be able to do it in a single Handler instead of multiple HandlerFunc.

Reply:
[commit 4](https://github.com/virtual-kubelet/virtual-kubelet/pull/935/commits/92d1bd4428dcfabc7456791a6f63d93caa06db57)  has changed to a single http.Handler but I recommend to keep this type and `func Middlewares` because we will have at least 2 middlewares (InstrumentHandler, and AuthFilter) for that handler. Please take a look. @sargun  @cpuguy83

Reply:
I'm not sure I understand the need, since a Handler is sufficient to wrap another handler, whereas a `Middleware` is a function to create a handler from a handler. I don't think we need a type for this, or a list of them.

We can conceivably provide a function like `func WrapHandler(http.Handler) http.Handler` to facilitate building an `http.Handler` that wraps.

Reply:
ok, this type is removed. please review again.